image/build_windows_containerdisk.sh

```diff
@@ -0,0 +1,94 @@
+#!/bin/bash
+
+set -e
+
+if [ -z "$1" ]; then
+  echo "Please pass Windows image version to build as first argument."
+  exit 1
+fi
+
+if [ -z "$2" ]; then
+  echo "Please pass path to Windows ISO as second argument."
+  exit 1
+fi
+
+WINDOWS_VERSION=$1
+WINDOWS_ISO=$2
+# EFI boot by default
+WINDOWS_BOOT_IMAGE=(-e efi/microsoft/boot/efisys_noprompt.bin)
+case "$WINDOWS_VERSION" in
+  windows10)
+    OS_VARIANT=win10
+    WINDOWS_ISO_OVERLAYS=(overlays/generic overlays/windows10)
+    VIRT_INSTALL_EXTRA_ARGS=(--boot uefi)
+    ;;
+  windows11)
+    OS_VARIANT=win11
+    WINDOWS_ISO_OVERLAYS=(overlays/generic overlays/windows11)
+    ;;
+  windows2k16)
+    OS_VARIANT=win2k16
+    WINDOWS_ISO_OVERLAYS=(overlays/windows2k16)
+    # Create BIOS bootable ISO
+    WINDOWS_BOOT_IMAGE=(-b boot/etfsboot.com)
+    ;;
+  windows2k19)
+    OS_VARIANT=win2k19
+    WINDOWS_ISO_OVERLAYS=(overlays/generic overlays/windows2k19)
+    # Create BIOS bootable ISO
+    WINDOWS_BOOT_IMAGE=(-b boot/etfsboot.com)
+    ;;
+  windows2k22)
+    OS_VARIANT=win2k22
+    WINDOWS_ISO_OVERLAYS=(overlays/generic overlays/windows2k22)
+    VIRT_INSTALL_EXTRA_ARGS=(--boot uefi)
+    ;;
+  *)
+    echo "Need a valid Windows image version to build: windows10, windows11, windows2k16, windows2k19, windows2k22"
+    exit 1
+    ;;
+esac
+
+# Extract Windows ISO, create a bootable ISO with overlays applied that does not prompt for a key and clean up
+# By default the ISO is EFI bootable
+7z x "${WINDOWS_ISO}" -o"/tmp/${WINDOWS_VERSION}"
+mkisofs -udf --allow-limited-size "${WINDOWS_BOOT_IMAGE[@]}" -no-emul-boot \
+  -o "/tmp/${WINDOWS_VERSION}.iso" "/tmp/${WINDOWS_VERSION}" "${WINDOWS_ISO_OVERLAYS[@]}"
+rm -r "/tmp/${WINDOWS_VERSION}"
+
+# Download latest virtio-win.iso if it was not passed as argument, else copy passed ISO to /tmp
+if [ -z "$3" ]; then
+  curl -L https://fedorapeople.org/groups/virt/virtio-win/direct-downloads/latest-virtio/virtio-win.iso -o /tmp/virtio-win.iso
+else
+  cp "$3" /tmp/virtio-win.iso
+fi
```

Comment:
Maybe clean up the `/tmp/virtio-win.iso` file after competition?

Reply:
Also, you can store `/tmp/virtio-win.iso` and `/tmp/${WINDOWS_VERSION}` in a variable. It would make the code a bit cleaner.

Reply:
It is cleaned up by `virsh undefine --remove-all-storage`.

Reply:
Updated the PR.